SwiftUI/Model/SearchOperation/SearchOperation.mm

```diff
@@ -108,13 +111,14 @@ - (void)addTitleSearchResults:(std::vector<zim::Archive>)archives count:(int)cou
         auto results = zim::SuggestionSearcher(archive).suggest(self.searchText_C).getResults(0, count);
         for (auto result = results.begin(); result != results.end(); result++) {
             if (self.isCancelled) { break; }
-            
-            zim::Entry entry = result.getEntry();
-            zim::Item item = entry.getItem(entry.isRedirect());
-            NSString *path = [NSString stringWithCString:item.getPath().c_str() encoding:NSUTF8StringEncoding];
-            NSString *title = [NSString stringWithCString:item.getTitle().c_str() encoding:NSUTF8StringEncoding];
-            SearchResult *searchResult = [[SearchResult alloc] initWithZimFileID:zimFileID path:path title:title];
-            if (searchResult != nil) { [self.results addObject:searchResult]; }
+            NSString *path = [NSString stringWithCString:result->getPath().c_str() encoding:NSUTF8StringEncoding];
+            NSString *title = [NSString stringWithCString:result->getTitle().c_str() encoding:NSUTF8StringEncoding];
+            if (title.length > 0) {
```

Comment:
But there is no suggestion result for entries without a title… That's where using the path as title is the most important since suggestions are only presented as a Title with no excerpt or anything.

Reply:
Ok, I see. I did check this scenario as well. Currently if we display such item, the user will face an error trying to opening it. Therefore, I think it is better not to show it:

https://github.com/kiwix/apple/assets/6784320/212b9740-0dfc-4f7e-aa2e-9b07bdbfe674

Reply:
<img width="244" alt="Screenshot 2024-02-04 at 09 16 21" src="https://github.com/kiwix/apple/assets/6784320/13419557-926a-4279-89a7-832cfdd1a07e">

Reply:
OK then but what's the reason for it not loading? This looks like a separate issue that needs a ticket (we don't have to solve it now)

Reply:
> That's where using the path as title is the most important since suggestions are only presented as a Title with no excerpt or anything.
> OK then but what's the reason for it not loading? 

`Suggestion::getTitle` do not throw a exception, it returns an empty title.
So if the title is empty, it is probably because the entry has been removed. Using the path will not help to find the entry.

Reply:
@rgaudin Good for you there?